kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -531,6 +531,20 @@ internal class CodeWriter constructor(
     return null
   }
 
+  /**
+   * Returns the qualified name of the `simpleName` enum constant if it's in the current scope
+   */
+  private fun resolveEnumConstant(simpleName: String): String? {
```

Comment:
I made this into a separate function because I wasn't sure if a `ClassName` could represent an enum constant.

Reply:
It's odd that this method returns a different type, hence the check on line 410 can now fail because of the type check, and not logical inequality. We don't represent enum constants as `ClassName`s, so not sure what the outcome will be if we tried.